From 5a467c6b9b5dbe05676eaad55a3fbf917733cfd3 Mon Sep 17 00:00:00 2001
From: jordan Johnston <johnstonljordan@gmail.com>
Date: Mon, 15 Apr 2024 11:30:47 -0500
Subject: [PATCH] Render/Show All Layers

 -> Don't hide Top or Overlays layers

 NOTES:

 This does introduce an issue where you can interact with windows & 
 layers underneath Hyprspace, but in practice - I personally don't 
 care, atm. It doesn't really affect my use of it.

 IMPORTANT: for this  work as intended this needs to have: 
 
 `reservedArea = 36` (my AGS TopBar/Panel height) in Hyprspace's 
 plugin settings.

 Generally speaking, my hyprland.conf has all of the related plugin
 settings to integrate nicely with my n7n-AGS-Shell.

Signed-off-by: jordan Johnston <johnstonljordan@gmail.com>
---
 src/Overview.cpp          | 23 ----------
 2 files changed, 91 insertions(+), 23 deletions(-)
 create mode 100644 0001-static-win-pos.patch

diff --git a/src/Overview.cpp b/src/Overview.cpp
index ffb5339..b0e8991 100644
--- a/src/Overview.cpp
+++ b/src/Overview.cpp
@@ -44,15 +44,6 @@ void CHyprspaceWidget::show() {
         }
     }
 
-    // hide top and overlay layers
-    // FIXME: ensure input is disabled for hidden layers
-    for (auto& ls : owner->m_aLayerSurfaceLayers[ZWLR_LAYER_SHELL_V1_LAYER_TOP]) {
-        ls->startAnimation(false);
-    }
-    for (auto& ls : owner->m_aLayerSurfaceLayers[ZWLR_LAYER_SHELL_V1_LAYER_OVERLAY]) {
-        ls->startAnimation(false);
-    }
-
     active = true;
 
     // swiping panel offset should be handled at updateSwipe
@@ -70,20 +61,6 @@ void CHyprspaceWidget::hide() {
     auto owner = getOwner();
     if (!owner) return;
 
-    // restore layer state
-    for (auto& ls : owner->m_aLayerSurfaceLayers[ZWLR_LAYER_SHELL_V1_LAYER_TOP]) {
-        if (!ls->readyToDelete && ls->mapped) {
-            ls->fadingOut = false;
-            ls->startAnimation(true);
-        }
-    }
-    for (auto& ls : owner->m_aLayerSurfaceLayers[ZWLR_LAYER_SHELL_V1_LAYER_OVERLAY]) {
-        if (!ls->readyToDelete && ls->mapped) {
-            ls->fadingOut = false;
-            ls->startAnimation(true);
-        }
-    }
-
     if (!swiping) {
         // restore fullscreen state
         for (auto& fs : prevFullscreen) {
-- 
2.44.0
